lms/djangoapps/courseware/module_render.py

```diff
@@ -475,13 +476,15 @@ def publish(block, event_type, event):
                 only_if_higher=event.get('only_if_higher'),
             )
         else:
-            aside_context = {}
+            context = contexts.course_context_from_course_id(course_id)
+            context['user_id'] = block.runtime.user_id
+            context['asides'] = {}
             for aside in block.runtime.get_asides(block):
                 if hasattr(aside, 'get_event_context'):
                     aside_event_info = aside.get_event_context(event_type, event)
                     if aside_event_info is not None:
-                        aside_context[aside.scope_ids.block_type] = aside_event_info
-            with tracker.get_tracker().context('asides', {'asides': aside_context}):
+                        context['asides'][aside.scope_ids.block_type] = aside_event_info
+            with tracker.get_tracker().context(event_type, context):
```

Comment:
Another example event emitted from the same module_render.py code:

`{"username": "d98e342511a445b184ccdc5835a3a9", "event_type": "problem_check", "ip": "10.0.2.2", "agent": "python-requests/2.12.4", "host": "precise64", "referer": "http://localhost:8000", "accept_language": "", "event": {"submission": {"0d759dee4f9d459c8956136dbde55f02_2_1": {"input_type": "textline", "question": "", "group_label": "", "response_type": "stringresponse", "answer": "WRONG", "variant": "", "correct": false}}, "success": "incorrect", "grade": 0, "correct_map": {"0d759dee4f9d459c8956136dbde55f02_2_1": {"hint": "", "hintmode": null, "correctness": "incorrect", "npoints": null, "answervariable": null, "msg": "", "queuestate": null}}, "state": {"correct_map": {"0d759dee4f9d459c8956136dbde55f02_2_1": {"hint": "", "hintmode": null, "correctness": "incorrect", "npoints": null, "answervariable": null, "msg": "", "queuestate": null}}, "input_state": {"0d759dee4f9d459c8956136dbde55f02_2_1": {}}, "has_saved_answers": false, "seed": 1, "done": true, "student_answers": {"0d759dee4f9d459c8956136dbde55f02_2_1": "WRONG"}}, "answers": {"0d759dee4f9d459c8956136dbde55f02_2_1": "WRONG"}, "attempts": 55, "max_grade": 1, "problem_id": "block-v1:edX+DemoX+Demo_Course+type@problem+block@0d759dee4f9d459c8956136dbde55f02"}, "event_source": "server", "context": {"course_user_tags": {}, "user_id": 339, "org_id": "edX", "asides": {}, "module": {"usage_key": "block-v1:edX+DemoX+Demo_Course+type@problem+block@0d759dee4f9d459c8956136dbde55f02", "display_name": "Text Input"}, "course_id": "course-v1:edX+DemoX+Demo_Course", "path": "/courses/course-v1:edX+DemoX+Demo_Course/xblock/block-v1:edX+DemoX+Demo_Course+type@problem+block@0d759dee4f9d459c8956136dbde55f02/handler/xmodule_handler/problem_check"}, "time": "2017-01-19T20:20:03.486545+00:00", "page": "x_module"}`